internal/gengapic/gengapic.go

```diff
@@ -619,11 +619,26 @@ func (g *generator) insertMetadata(m *descriptor.MethodDescriptorProto) error {
 			}
 			seen[field] = true
 
+			accessor := fmt.Sprintf("req%s", buildAccessor(field))
+			typ := g.lookupFieldType(m.GetInputType(), field)
+
+			// TODO(noahdietz): need to handle []byte for TYPE_BYTES.
+			if typ == descriptor.FieldDescriptorProto_TYPE_STRING {
+				accessor = fmt.Sprintf("url.QueryEscape(%s)", accessor)
+			} else if typ == descriptor.FieldDescriptorProto_TYPE_DOUBLE || typ == descriptor.FieldDescriptorProto_TYPE_FLOAT {
+				// Format the floating point value with mode 'g' to allow for
+				// exponent formatting when necessary, and decimal when adequate.
+				// QueryEscape the resulting string in case there is a '+' in the
+				// exponent.
+				// See golang.org/pkg/fmt for more information on formatting.
+				accessor = fmt.Sprintf("url.QueryEscape(fmt.Sprintf(%q, %s))", "%g", accessor)
+			}
+
 			// URL encode key & values separately per aip.dev/4222.
 			// Encode the key ahead of time to reduce clutter
 			// and because it will likely never be necessary
-			fmt.Fprintf(&values, " %q, url.QueryEscape(req%s),",
-				url.QueryEscape(field), buildAccessor(field))
+			fmt.Fprintf(&values, " %q, %s,", url.QueryEscape(field), accessor)
+			// Produces either "%s=%v&" or "%s=%f&" depending on the field type.
```

Comment:
I'm confused by this comment: the string below is not conditional

Reply:
Oops it was a comment I forgot to delete when changing the code. Deleted!